app/jobs/export_pool_xp_to_traction_job.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+# Combine the samples from a Pool XP tube into a compound sample, generate a bioscan-pool-xp-tube-to-traction message
+# and submit it to RabbitMQ so that it can forwarded to Traction by the message processor.
+ExportPoolXpToTractionJob =
+  Struct.new(:barcode) do
+    include CompoundSampleHelper
+
+    def perform
+      message_data = get_message_data(barcode)
+
+      subject_obj = configatron.amqp.schemas.subjects[:export_pool_xp_to_traction]
+      subject = subject_obj[:subject]
+      version = subject_obj[:version]
+
+      schema = get_message_schema(subject, version)
+      encoded_message = avro_encode_message(message_data, schema)
+      send_message(encoded_message, subject, version)
+    rescue StandardError => e
+      Rails.logger.error("Error exporting Pool XP tube to Traction: <#{e.message}>")
+      raise
+    end
+
+    def fetch(uri_str, limit = 10)
```

Comment:
Might be worth giving this a slightly more specific name. `fetch` is a method on a `Hash` in Ruby, so could be a little confusing.

Reply:
Perhaps `fetch_response` since the method on `HTTP` is `get_response`?